### ISSUE TITLE
Easily connect / disconnect from multiple VPNs

### ISSUE THREAD

**Issue by lonkelle** [dperson/openvpn-client#333]
I have 10 VPNs I'd like to pick and choose from. I got one of them working, but had to do that at build time to include the file in the docker. Is there an easier way to choose between 10 different OVPN files (all with fully embedded certs, no username / password needed)?

**Comment by gynnantonix** [comment]
Are you asking how to programmatically choose a random file from a given set of files?

**Comment by lonkelle** [comment]
That is exactly correct. I also, at this point, only know how to inject OVPN files at build time. So I have to re-build every time I want to connect to a new OVPN file I get, if I knew a way to overwrite the file without re-building, that would be even better. 

Thank you for commenting. I’m been scratching my head about this for a week now (before I made an issue on here) trying to figure it out. I’m also **very** new to Docker so that could be why I’m not realizing the, very likely, simple solution. 😬😅

**Comment by gynnantonix** [comment]
Does this mean that you're already building your `.ovpn` files into `/configs/<your_config_dir>` in the VPN container?  If so, then the next step is to include a `start.sh` script in `<your_config_dir>` that selects a random configuration file from the current directory and links it to `vpn.conf`.

**Comment by lonkelle** [comment]
Yes, I have ten pre-built .ovpn files. And I can't figure out how to copy more than one of them into the docker container. This is how I currently do it and is the only thing that has actually gotten the container to connect to any VPN after about 50 different experiments:

`cd /docker-openvpn-client`
`docker build -t dperson/openvpn-client:latest .`
`docker run -it --cap-add=NET_ADMIN --device /dev/net/tun --name vpn --dns 8.8.8.8 -d dperson/openvpn-client`

That works perfectly, but only because I renamed one of my .ovpn files to vpn.conf and docker picks it up in the build process and copies it over to a `/vpn` folder inside of Docker. So if I want to change files / VPNs, I have to replace the vpn.conf and re-build.

I'm new to Docker entirely as of this week so I must sound so uninformed (and might be missing something very simple), but I got it working with one VPN and I've re-read the documentation over and over and experimented over and over and can't figure out how to include **all** of my pre-configured .ovpn files as well **and then** select one of them on `docker run`. Here's a screenshot of my folder with my renamed .ovpn file (to vpn.conf):

<img width="580" alt="Screen Shot 2020-09-14 at 3 20 50 PM" src="https://user-images.githubusercontent.com/5572600/93129537-2a695300-f69f-11ea-9f9d-e0f610e2c25c.png">

PS. Thank you for your comments, you have no idea how much of a relief it is to talk to someone in-the-know instead of trying to continue to figure this out on my own like I've been doing.

**Comment by gynnantonix** [comment]
I'm going to put up a repo for you that contains the required changes

**Comment by gynnantonix** [comment]
Try this: https://github.com/gynnantonix/openvpn-client
Just dump your `*.ovpn` config files into the `configs` dir, along with `vpn.auth` containing your VPN username and password on separate lines, build the container, and the image should randomly pick one of your config files and run with it.
If you'd like to modify the selection process, make your changes in `configs/configure.sh`.

**Comment by lonkelle** [comment]
Thanks for the recommendation and instruction, you're a lifesaver - seriously, have been scratching my head on this for so long! I'll try that right now. Is there a way to inject a new `*.ovpn` files in an already existing docker container or would I always have to "re-build" the container to do so? Oh, and none of my `*.ovpn` files have a username or password, do you think I need to do anything with `vpn.auth` to indicate that?

**Comment by lonkelle** [comment]
Oh, I just realized, that is your personal fork of the openvpn-client Docker. Cool, thanks for pointing me your way! What features did you / are you planning to add?

**Comment by lonkelle** [comment]
I just checked your commits, and you **literally** forked and added that feature for me today. Thank you! That was really really kind. :)

**Comment by gynnantonix** [comment]
I'm glad to help :)

**Comment by lonkelle** [comment]
@gynnantonix I can't seem to find a way to disconnect, and reconnect to the different `.ovpn` files I provided with your fork without quitting the docker container and restarting it with a different selected `.ovpn` file. Do you think there's a way to disconnect, and re-connect **without** stopping and then starting the container over?

**Comment by OptiMous28** [comment]
is this feature something that can be configured in this project? say, cycling every 12 hours between different ovpn configs without the need to rebuild/restart the docker container.

**Comment by lonkelle** [comment]
I built the app with on-demand in mind. I got it working with a full PHP backend to change OVPN configs. It changes instantly **but** a restart of the container is necessary. I wish there was a way to disconnect and connect on demand without switching.

**Comment by Peppershaker** [comment]
thanks @gynnantonix
